### ISSUE TITLE
[Bug]: TypeError: [html-rspack-plugin]: Cannot read properties of undefined (reading 'content')

### ISSUE THREAD

**Issue by beilo** [web-infra-dev/rspack#4748]
### 版本信息

```sh
System:
    OS: macOS 12.5.1
    CPU: (8) x64 Intel(R) Core(TM) i5-8257U CPU @ 1.40GHz
    Memory: 1.18 GB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Browsers:
    Chrome: 118.0.5993.88
    Safari: 15.6.1
    Safari Technology Preview: 16.0
  npmPackages:
    @rsbuild/core: ^0.0.24 => 0.0.24 
    @rsbuild/plugin-react: ^0.0.24 => 0.0.24
```

### 问题详情

当我设置了maxSize就会报错
When I set maxSize, an error will occur.
<img width="324" alt="image" src="https://github.com/web-infra-dev/rsbuild/assets/19225489/f21c0bff-1f7a-4139-a02b-1e9c120afaf2">

<img width="1013" alt="image" src="https://github.com/web-infra-dev/rsbuild/assets/19225489/c81e7139-0222-4516-aeef-41b8fcb07d68">

### 复现链接

img

### 复现步骤

error[internal]: TypeError: [html-rspack-plugin]: Cannot read properties of undefined (reading 'content')
/node_modules/html-webpack-plugin/index.js:373:81
    at async Promise.all (index 2)
    at async Promise.all (index 1)

**Comment by chenjiahan** [comment]
I have created a minimal repo: https://github.com/chenjiahan/rspack-repro-split-chunk-max-size

Repro steps:

1. pnpm i
2. pnpm dev:rspack

**Comment by chenjiahan** [comment]
@bvanjoi If you have time, please help to check this~

**Comment by bvanjoi** [comment]
It appears that ﻿`@rspack/core@0.4.1` operates effectively.

**Comment by chenjiahan** [comment]
Fixed in Rspack v0.4.1

**Comment by mavridiSS** [comment]
This bug still appears in v0.4.3
